tests/utils/test_dependency_specification.py

```diff
@@ -119,8 +147,13 @@ def _mock(self: Path) -> bool:
 
     mocker.patch("pathlib.Path.exists", _mock)
 
-    assert not DeepDiff(
-        RequirementsParser(artifact_cache=artifact_cache).parse(requirement),
-        specification,
-        ignore_order=True,
+    if isinstance(specification, dict):
+        specification = [specification]
```

Comment:
Could you just change the data in parametrize so we can avoid this?

Reply:
No strong feelings about this. I changed it in a separate commit. (In the end we'll squash the commits anyway.)